src/native/fs/fs_napi.cpp

```diff
@@ -186,41 +233,92 @@ struct Stat : public FSWorker
-        double ctimeMs = (double(1e3) * _stat_res.st_ctimespec.tv_sec) + (double(1e-6) * _stat_res.st_ctimespec.tv_nsec);
-        double mtimeMs = (double(1e3) * _stat_res.st_mtimespec.tv_sec) + (double(1e-6) * _stat_res.st_mtimespec.tv_nsec);
-        double birthtimeMs = (double(1e3) * _stat_res.st_birthtimespec.tv_sec) + (double(1e-6) * _stat_res.st_birthtimespec.tv_nsec);
-#else
-        double atimeMs = (double(1e3) * _stat_res.st_atim.tv_sec) + (double(1e-6) * _stat_res.st_atim.tv_nsec);
-        double ctimeMs = (double(1e3) * _stat_res.st_ctim.tv_sec) + (double(1e-6) * _stat_res.st_ctim.tv_nsec);
-        double mtimeMs = (double(1e3) * _stat_res.st_mtim.tv_sec) + (double(1e-6) * _stat_res.st_mtim.tv_nsec);
-        double birthtimeMs = ctimeMs; // Posix doesn't have birthtime
-#endif
+        res["type"] = Napi::Number::New(env, _statfs_res.f_type);
+        res["bsize"] = Napi::Number::New(env, _statfs_res.f_bsize);
+        res["blocks"] = Napi::Number::New(env, _statfs_res.f_blocks);
+        res["bfree"] = Napi::Number::New(env, _statfs_res.f_bfree);
+        res["bavail"] = Napi::Number::New(env, _statfs_res.f_bavail);
+        res["files"] = Napi::Number::New(env, _statfs_res.f_files);
+        res["ffree"] = Napi::Number::New(env, _statfs_res.f_ffree);
+        // Linux, SunOS, HP-UX, 4.4BSD have a system call statfs() that returns a struct statfs (defined in <sys/vfs.h>) 
+        // containing a fsid_t f_fsid, where fsid_t is defined as struct { int val[2]; }.
+        // so it's an array of two integers
+        // For now commenting out res["fsid"] as we cant use Napi::Number on it
+        // TODO: when we will need fsid in the results uncomment it and fix it.
+        // res["fsid"] = Napi::Number::New(env, _statfs_res.f_fsid);
+        res["flags"] = Napi::Number::New(env, _statfs_res.f_flags);
 
-        res["atimeMs"] = Napi::Number::New(env, atimeMs);
-        res["ctimeMs"] = Napi::Number::New(env, ctimeMs);
-        res["mtimeMs"] = Napi::Number::New(env, mtimeMs);
-        res["birthtimeMs"] = Napi::Number::New(env, birthtimeMs);
-        res["atime"] = Napi::Date::New(env, uint64_t(round(atimeMs)));
-        res["mtime"] = Napi::Date::New(env, uint64_t(round(mtimeMs)));
-        res["ctime"] = Napi::Date::New(env, uint64_t(round(ctimeMs)));
-        res["birthtime"] = Napi::Date::New(env, uint64_t(round(birthtimeMs)));
+        #ifdef __APPLE__   
+            //  res["otype"] =  Napi::Number::New(env, _statfs_res.f_otype);
+            //  res["oflags"] =  Napi::Number::New(env, _statfs_res.f_oflags);
+            //  res["iosize"] =  Napi::Number::New(env, _statfs_res.iosize);
+            res["owner"] =  Napi::Number::New(env, _statfs_res.f_owner);
+            //  res["reserved1"] =  Napi::Number::New(env, _statfs_res.f_reserved1);
+        #else
+            res["namelen"] = Napi::Number::New(env, _statfs_res.f_namelen);
+            res["frsize"] = Napi::Number::New(env, _statfs_res.f_frsize);
+        #endif
```

Comment:
Are we going to use any of those OS specific properties??
I hope not, and I would remove those to avoid confusion and errors between developer env and runtime env.